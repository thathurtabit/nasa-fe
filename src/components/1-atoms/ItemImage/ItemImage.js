import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';
import IMG from './ItemImage.styled';
import LoadingSmall from '../LoadingSmall/LoadingSmall';
import { loadDelay } from '../../../utils/constants/constants';

export default class ItemImage extends Component {
  constructor(props) {
    super(props);
    this.state = { isLoading: true };
  }

  componentDidMount() {
    // Only show images once load has complete
    // Note: Must be a prettier way of doing this
    const img = new Image();
    const { url } = this.props;
    img.src = url;
    img.onload = () =>
      setTimeout(() => this.setState({ isLoading: false }), 100);
  }

  render() {
    const { isLoading } = this.state;
    const { title, url, thumb } = this.props;
    return (
      <Fragment>
        {isLoading && <LoadingSmall />}
        {!isLoading && (
          <CSSTransition
            in={!isLoading}
            classNames="thumb"
            appear
            timeout={loadDelay}
          >
            <IMG className="thumb" src={url} alt={title} thumb={thumb} />
          </CSSTransition>
        )}
      </Fragment>
    );
  }
}

ItemImage.propTypes = {
  thumb: PropTypes.bool,
  url: PropTypes.string.isRequired,
  title: PropTypes.string,
};

ItemImage.defaultProps = {
  thumb: false,
  title: 'Item',
};
