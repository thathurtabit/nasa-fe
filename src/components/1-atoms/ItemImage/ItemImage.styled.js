import styled from 'styled-components';

const IMG = styled.img`
  border: ${props =>
    props.thumb ? `2px solid ${props.theme.colors.bodyBg}` : '0'};
  border-top: 0;
  border-left: 0;
  box-shadow: ${props =>
    props.thumb ? '5px 5px 0 rgba(0, 0, 0, 0.05)' : 'none'};
  display: block;
  margin: 0 auto;
  max-width: 100%;
  max-height: 100%:
  opacity: 0;
  transform: scale(0.95);
  transition: box-shadow ${props => props.theme.thumb.transition.duration}
      ${props => props.theme.transition.easeOut},
    transform ${props => props.theme.thumb.transition.duration}
      ${props => props.theme.transition.easeOutBack},
    opacity ${props => props.theme.thumb.transition.duration}
      ${props => props.theme.transition.easeOut};
  width: 200px;

  @media screen and (min-width: ${props => props.theme.breakpoints.md}) {
    width: ${props => (props.thumb ? '250px' : '100%')};
  }

  a:focus &,
  &:hover {
    box-shadow: ${props =>
      props.thumb ? `10px 10px 0 ${props.theme.colors.brand}` : 'none'};
  }

  &.thumb-enter-done {
    opacity: 1;
    transform:  scale(1);
  }
`;

export default IMG;
