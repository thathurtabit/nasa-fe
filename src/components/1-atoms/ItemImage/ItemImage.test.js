import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import 'jest-styled-components';
import { shallowWithTheme } from '../../../utils/test_config/testHelpers';

import ItemImage from './ItemImage';

const title = 'test title';
const url = 'https://images-assets.nasa.gov/image/PIA12235/PIA12235~thumb.jpg';

describe('ItemImage', () => {
  it('should render correctly', () => {
    const element = shallowWithTheme(<ItemImage url={url} title={title} />);
    expect(shallowToJson(element)).toMatchSnapshot();
  });
});
