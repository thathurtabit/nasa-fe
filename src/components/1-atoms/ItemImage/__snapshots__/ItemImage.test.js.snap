// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemImage should render correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
